src/components/ActionButton/ToggleActionButton.ts

```diff
@@ -12,6 +12,44 @@ export interface IToggleActionButtonOptions {
 }
 
 export class ToggleActionButton extends Component {
+    /**
+     * Create the deactivated state for a given ToggleActionButton
+     * @param button {ToggleActionButton}
+     */
+    static generateDeactivatedStateInstance(button: ToggleActionButton): IStatefulActionButtonState {
```

Comment:
Why are these methods exposed? I would make them private instance methods instead.

Reply:
They could, but given what'll be coming in louis-bompart/search-ui-extensions#2 I don't think it's a matter for concerns because state definitions will be split from their use cases for code clarity.